packages/nimble-components/src/table/components/row/index.ts

```diff
@@ -96,6 +97,18 @@ export class TableRow<
         this.emitToggleEvent('row-action-menu-toggle', event.detail, column);
     }
 
+    public closeOpenActionMenus(): void {
+        if (this.menuOpen) {
+            const cellWithMenuOpen = Array.prototype.find.call(
+                this.shadowRoot!.children,
+                c => c instanceof TableCell && c.menuOpen
+            ) as TableCell;
+            if (cellWithMenuOpen?.actionMenuButton) {
+                cellWithMenuOpen.actionMenuButton.open = false;
```

Comment:
I'm noticing in the storybook events that I'm not getting before-toggle events as it closes:

When I normally interact with the control by clicking the menu and clicking an item I get the beforetoggle+toggle pairs, when I click the menu and then scroll the grid to close it via mouse wheel I don't see the beforetoggle

![image](https://user-images.githubusercontent.com/1588923/228984071-67a4a272-0508-4044-a6e5-b372baac04b5.png)

Reply:
@mollykreis do you think the `beforetoggle` event should fire in this case?
Currently [only the `private setOpen()` method in the menu button triggers the `beforetoggle` event](https://github.com/ni/nimble/blob/523d60f92bf266d9ecee680762d7ddd606638af8/packages/nimble-components/src/menu-button/index.ts#L193), setting `open` directly/programmatically does not.

Reply:
Yes, we should probably be consistent and fire the event in that case as well.

Reply:
Not sure the cleanest way to accomplish this without changing the semantics of the `beforetoggle` event (i.e. that it should fire before `open` changes, not after).
- could make MenuButton `setOpen()` public - this works, but we're not really enforcing that `beforetoggle` would always fire then (i.e. it's still true that setting `open` directly would not trigger `beforetoggle`)
- I thought about firing `beforetoggle` in a property setter for `open`, but then I couldn't use `@attr` anymore
- fire both `beforetoggle` and `toggle` in `openChanged()` - but changes when the event is fired as I mentioned

Reply:
Another issue I think I'm seeing is that the menu-button doesn't seem to fire `beforetoggle` events at all during programmatic changes (probably for similar reasons, it would have to happen during property assignment before the template binding updates).

The intention of beforetoggle in the popover api is that the operation is cancellable (in addition to having the right timing to populate stuff before it actually is shown). That makes sense in that api with the [show/hide/toggle-Popover functions](https://html.spec.whatwg.org/multipage/popover.html#dom-showpopover). Those functions also better handle situations like erroring when you can't actually open (like when not attached to the DOM).

I think we need a beforetoggle event even if the timing is not correct (for the contract for folks populating the action menu contents programmatically). Maybe we can get away with having beforetoggle and toggle firing back-to-back for the action menu when closing. It's important to get befoeretoggle to populate your DOM before the menu is shown, but I don't think it's as important to clear your DOM before it is hidden (maybe that should even be actively discouraged?...).

Proposal (@mollykreis @msmithNI ):
- On action menu close fire the beforetoggle and toggle back to back for the event lifecycle symmetry.
- Create an issue for the beforetoggle not firing programmatically for the menu-button. Capture some options:
  - Create setters with event firing side-effects that are cancellable?
  - fork from FAST's API and migrate to an API that better aligns with the popover api.
- Create an issue for the incorrect lifecycle behavior of beforetoggle for the action menu.

Reply:
Filed #1157 to track the `beforetoggle` event issues.  
For the purposes of this PR, I updated the table virtualizer code to close the action menu via a click on the MenuButton's toggle button, as that results in `beforetoggle` firing like we want, but doesn't need API/ behavior changes to the MenuButton for now.